quint/src/cliCommands.ts

```diff
@@ -331,6 +335,37 @@ export async function runTests(prev: TypecheckedStage): Promise<CLIProcedure<Tes
       }
     },
   }
+
+  // Find tests that are not used in the main module. We need to add references to them in the main module so flattening
+  // doesn't drop the definitions.
+  const unusedTests = [...testing.unusedDefinitions(mainName)].filter(
+    d => d.kind === 'def' && options.testMatch(d.name)
+  )
+
+  // Define name expressions referencing each test that is not referenced elsewhere, adding the references to the lookup
+  // table
+  const args: QuintEx[] = unusedTests.map(defToAdd => {
+    const id = testing.idGen.nextId()
+    testing.table.set(id, defToAdd)
+    const namespace = defToAdd.importedFrom ? qualifier(defToAdd.importedFrom) : undefined
+    const name = namespace ? [namespace, defToAdd.name].join('::') : defToAdd.name
```

Comment:
This seems like a slightly convoluted conditional test, that has to go through whether the temp variable `namespace` is undefined? iiuc, the point of the two lines, iiuc, is that the name should be qualified if `detToAdd.importedFrom` is truthy. This logic could be expressed a bit more directly, imo, via something like

```suggestion
    const name = defToAdd.importedFrom ? [qualifier(defToAdd.importedFrom), defToAdd.name].join('::') : defToAdd.name
```

Reply:
> iiuc, is that the name should be qualified if detToAdd.importedFrom is truthy.

This is a correct if but not iff. `qualifier()` might return undefined as well, since not every import definition introduces a qualifier. So in your suggestion, we'd have to add an additional check for undefined, which I believe will make it more convoluted.